### PR TITLE
feat(controlplane): filter workflow runs by status

### LINKED ISSUE #545: feat: allow filtering workflow runs by status
To be able to filter workflow runs by `status`. Eventually we will also allow to filter by runner and so on.

### PR DESCRIPTION
**Pull request by migmartri**
Adds support in the controlplane API to filter workflow runs by `status`

For example

```console
$ grpcurl --plaintext -H "authorization: Bearer $TOKEN" -d '{"status": "RUN_STATUS_SUCCEEDED"}' localhost:9000 controlplane.v1.WorkflowRunService.List   
```
```json
{
  "result": [
    {
      "id": "fe015f2d-b705-4df1-beb2-751f29878c34",
      "createdAt": "2024-02-28T21:09:36.087997Z",
      "finishedAt": "2024-02-28T21:09:43.042305Z",
      "state": "success",
      "workflow": {
        "id": "b63fddfd-ca24-4623-b3b8-700ad26368e5",
        "name": "foo",
        "project": "faz",
        "createdAt": "2024-02-28T21:09:07.944265Z",
        "runsCount": 4,
        "description": "bar"
      },
      "contractRevisionUsed": 1,
      "contractRevisionLatest": 1,
      "status": "RUN_STATUS_SUCCEEDED"
    }
  ],
  "pagination": {
    
  }
}

```

In addition to this change, the API now is starting to return a strongly typed `run status` and deprecated the previous one

```
"status": "RUN_STATUS_SUCCEEDED" // replaces `state`
```

This is important since the filter by state is strongly typed as well.

There are some other minor cleanup related to issues raised by the linter.

Closes #545 

cc/ @tknura

### CODE REVIEW

File: app/cli/cmd/auth_login.go [hunk 1]
```diff
@@ -85,9 +85,6 @@ func interactiveAuth() error {
 
 	// Run server in background
 	http.HandleFunc(callbackURL.Path, a.handleCallback)
-	if err != nil {
```
Comment (migmartri) [inline app/cli/cmd/auth_login.go:88]: this is code that wasn't been used and the linter complained about.

File: app/cli/cmd/workflow_workflow_run_describe.go [hunk 2]
```diff
@@ -198,7 +198,7 @@ func encodeAttestationOutput(run *action.WorkflowRunItemFull) error {
 	}
 
 	// It could not be encoded but for a reason that's not because it was a custom format
-	if err == nil && !errors.Is(err, ErrOutputFormatNotImplemented) {
```
Comment (migmartri) [inline app/cli/cmd/workflow_workflow_run_describe.go:201]: same about this impossible condition 🤦🏼

File: app/controlplane/internal/biz/workflowrun.go [hunk 3]
```diff
@@ -256,22 +256,19 @@ func (uc *WorkflowRunUseCase) SaveAttestation(ctx context.Context, id string, en
 	return uc.wfRunRepo.SaveAttestation(ctx, runID, envelope, digest)
 }
 
+type RunListFilters struct {
```
Comment (migmartri) [inline app/controlplane/internal/biz/workflowrun.go:259]: this filter object will help us extend this call in the future i.e by filtering by runner type.

**Review by danlishka — APPROVED** [review]
LGTM

**Pull request merged**